channels/applicationinsights-channel-js/src/Sender.ts

```diff
@@ -264,6 +269,11 @@ export class Sender extends BaseTelemetryPlugin implements IChannelControlsAI {
                         }
                     }
                 }
+
+                // initialize statsbeat instance last after sender config is fully populated
+                if (_self._statsbeat && config.disableStatsbeat !== true) {
```

Comment:
Why even create the `Statsbeat` object if config is set to disabled? Why can't initializing happen on creation instead of explicitly calling `initialize`? Is this a design pattern used across the sdk?

Reply:
The design is to pass a statsbeat instance through sender and use that for each AI. And statsbeat itself has a sender as well, we figured out this way to differentiate these two senders by checking if a statsbeat instance is passed in.